### PR TITLE
[Improvement][Dao] SQL optimization - check exist #5069

### LINKED ISSUE #5069: [Improvement][Dao] SQL optimization - check exist
**Describe the question**
As report by #5061, we need to optimize the sql.

**What are the current deficiencies and the benefits of improvement**
Improve the performance about check some exist with cleaner method

**Which version of DolphinScheduler:**
 -[dev]
 -[1.3.6]

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of the pull request
See #5061 

this closes #5069 

## Brief change log

Using select 1 from xx limit 1 to check the record exist.

## Verify this pull request
Add Unit test for each changed mapper

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/QueueServiceImpl.java [hunk 1]
```diff
@@ -293,7 +294,7 @@ private boolean checkQueueNameExist(String queueName) {
      * @return true if need to update user
      */
     private boolean checkIfQueueIsInUsing (String oldQueue, String newQueue) {
-        return !oldQueue.equals(newQueue) && CollectionUtils.isNotEmpty(userMapper.queryUserListByQueue(oldQueue));
+        return !oldQueue.equals(newQueue) && BooleanUtils.isNotTrue(userMapper.existUser(oldQueue));
     }
 
 }
```
Comment (CalvinKirs) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/QueueServiceImpl.java]: There is a problem here

Comment (ruanwenjun) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/QueueServiceImpl.java]: Thanks, this bug has been solved.

**Comment by CalvinKirs** [comment]
image
looks good to me, please remove unused import and resolved code style fail.

**Comment by ruanwenjun** [comment]
> image
> looks good to me, please remove unused import and resolved code style fail.

done ^^

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**